### PR TITLE
Reactive slicer

### LINKED ISSUE #64: Make slicer use reactive data.

### PR DESCRIPTION
**Pull request by mateuszwyszynski**
This resolves #64 

Slicer module is now works with data in form of reactive value. This is necessary if we want to be able to deal with data changing when the app is running (e.g. underlaying polygons change).

Moreover, use of slicer module is simplified - everything is done by server part of slicer module and therefore user does not have to worry about the ui function call.

TODO: Improve documentation.

### CODE REVIEW

File: R/slicer.R [hunk 1]
```diff
@@ -11,6 +11,38 @@ getTableOfSubtables <- function(dataTable, categoryName) {
   data.table(category = categories, dataTable = dataTables)
 }
 
+whichUiToRender <- function(ui, subtables, ns) {
+  switch(ui,
+         "tab" = {
+           tabPanelWithSlicerContent <- function(category, dataTable) {
```
Comment (DamianRodziewicz) [inline R/slicer.R]: dataTable is not used here

File: R/slicer.R [hunk 2]
```diff
@@ -11,6 +11,38 @@ getTableOfSubtables <- function(dataTable, categoryName) {
...
+           tabPanels <- pmap(subtables, function(category, dataTable) {
+             tabPanelWithSlicerContent(category, dataTable)
+           })
+
+           mainPanel(
+             do.call(tabsetPanel, tabPanels)
+           )
+         },
+         "box" = {
+           boxWithSlicerContent <- function(category, dataTable) {
```
Comment (DamianRodziewicz) [inline R/slicer.R]: dataTable is not used here

File: R/slicer.R [hunk 3]
```diff
@@ -11,6 +11,38 @@ getTableOfSubtables <- function(dataTable, categoryName) {
   data.table(category = categories, dataTable = dataTables)
 }
 
+whichUiToRender <- function(ui, subtables, ns) {
```
Comment (DamianRodziewicz) [inline R/slicer.R]: Rename to `generateUI`, rename `ui` to `uiType`

File: R/slicer.R [hunk 4]
```diff
@@ -11,6 +11,38 @@ getTableOfSubtables <- function(dataTable, categoryName) {
...
+
+           mainPanel(
+             do.call(tabsetPanel, tabPanels)
+           )
+         },
+         "box" = {
+           boxWithSlicerContent <- function(category, dataTable) {
+             shinydashboard::box(
+               width = 6, solidHeader = TRUE, collapsible = TRUE,
+               title = category, background = "light-blue",
```
Comment (DamianRodziewicz) [inline R/slicer.R]: Light-blue background doesn't look good. Please revert to white.

**Review by DamianRodziewicz — APPROVED** [review]
Three small remarks, other than that LGTM

File: R/slicer.R (hunk 3 above)
Comment (mateuszwyszynski) [reply R/slicer.R]: 👍

File: R/slicer.R (hunk 4 above)
Comment (mateuszwyszynski) [reply R/slicer.R]: 👍

File: R/slicer.R (hunk 2 above)
Comment (mateuszwyszynski) [reply R/slicer.R]: 👍

File: R/slicer.R (hunk 1 above)
Comment (mateuszwyszynski) [reply R/slicer.R]: 👍

**Comment by mateuszwyszynski** [comment]
Comments addressed, PTAL.

**Review by DamianRodziewicz — APPROVED** [review]
Please fix style error and merge.

**Pull request merged**